hazelcast/src/test/java/com/hazelcast/internal/util/phonehome/PhoneHomeIntegrationTest.java

```diff
@@ -70,7 +71,32 @@
     public void initialise() {
         HazelcastInstance hz = createHazelcastInstance();
         node = getNode(hz);
-        phoneHome = new PhoneHome(node, "http://localhost:8080/ping");
+        phoneHome = new PhoneHome(node,
+                "http://localhost:8080/ping",
+                "http://localhost:8080/latest/meta-data",
+                "http://localhost:8080/metadata/instance/compute?api-version=2018-02-01",
+                "http://localhost:8080/metadata.google.internal");
+    }
+
+    public void stubUrls(String phoneHomeStatus, String awsStatus, String azureStatus, String gcpStatus) {
+
+        stubFor(get(urlPathEqualTo("/ping"))
+                .willReturn(phoneHomeStatus.equals("200")
+                        ? aResponse().withStatus(200)
+                        : aResponse().withFault(Fault.CONNECTION_RESET_BY_PEER)));
+        stubFor(get(urlPathEqualTo("/latest/meta-data"))
+                .willReturn(awsStatus.equals("200")
+                        ? aResponse().withStatus(200)
+                        : aResponse().withFault(Fault.CONNECTION_RESET_BY_PEER)));
+        stubFor(get(urlPathEqualTo("/metadata/instance/compute"))
+                .withQueryParam("api-version", equalTo("2018-02-01"))
+                .willReturn(azureStatus.equals("200")
+                        ? aResponse().withStatus(200)
+                        : aResponse().withFault(Fault.CONNECTION_RESET_BY_PEER)));
+        stubFor(get(urlPathEqualTo("/metadata.google.internal"))
+                .willReturn(gcpStatus.equals("200")
+                        ? aResponse().withStatus(200)
+                        : aResponse().withFault(Fault.CONNECTION_RESET_BY_PEER)));
```

Comment:
Please resolve these duplicate conditionals. Also, the method is called at 6 places, at 5 of them the parameter lists are the same - this is an other code smell. Please think about how to resolve these.